src/test/java/GreetzLoginFavouritesTest.java

```diff
@@ -0,0 +1,134 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Random;
+
+public class GreetzLoginFavouritesTest {
+
+    private WebDriver driver = null;
+
+    private ChromeDriver createDriverInstance() {
+        System.setProperty("webdriver.chrome.driver", "src/main/resources/chromedriver");
+        return new ChromeDriver();
+    }
+
+    private void testLoginImpl() throws InterruptedException {
+        driver.get("https://www.greetz.nl/auth/login");
+        Thread.sleep(4000);
+
+        WebElement loginFormElem = driver.findElement(By.id("loginForm"));
+        String email = "test_testing@greetz.com";
+        String pass = "testgreetz94";
+        loginFormElem.findElement(By.name("email")).sendKeys(email);
+        loginFormElem.findElement(By.name("password")).sendKeys(pass);
+        Thread.sleep(3000);
+        driver.findElement(By.id("login-cta")).click();
+        Thread.sleep(5000);
+        Assert.assertTrue(isLoggedIn("test"), "Failed to log in user " + email);
+        Thread.sleep(5000);
+    }
+
+    public boolean isLoggedIn(String firstName) {
+        try {
+            driver.findElement(By.xpath("//div[@class='header-message']//span[text()='Welkom " + firstName + "']"));
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    @Test
+    public void testGreetzLogin() throws InterruptedException {
+        driver = createDriverInstance();
+        try {
+            testLoginImpl();
+        } finally {
+            driver.quit();
+        }
+    }
+
+    @Test
+    public void testFavouritesFunctionality() throws InterruptedException {
+        driver = createDriverInstance();
+        try {
+            testLoginImpl();
+
+            driver.get("https://www.greetz.nl/ballonnen/denken-aan");
+            Thread.sleep(3000);
+
+            WebElement chosenItem = driver.findElement(By.xpath("//div[text()=\"Ballon 'op fleuren'\"]/ancestor::a/following-sibling::a/div"));
+            chosenItem.click();
+            String expectedItemTitle = driver.findElement(By.xpath("//a[@href='/ballonnen/detail/ballon--op-fleuren-/1142785578']/div[2]")).getText();
+            String expectedItemPrice = driver.findElement(By.xpath("//div[text()=\"Ballon 'op fleuren'\"]/ancestor::a//span/span")).getText();
+            Thread.sleep(3000);
+
+            WebElement myGreetz = driver.findElement(By.linkText("MyGreetz"));
+            myGreetz.click();
+            Thread.sleep(3000);
+
+            WebElement favorieten = driver.findElement(By.xpath("//div[@class=\"hamburgermenu profilehamburgermenu hamburgermenu_visible\"]//span[text()='Favorieten']"));
+            favorieten.click();
+            Thread.sleep(3000);
+
+            WebElement favItem = driver.findElement(By.xpath("//*[@class='favorite-item']"));
```

Comment:
There can be several items found by this locator. Please use such one which finds a unique element

Reply:
Addressed in 7968d48.